modules/gateways/blockonomics/usdtblockonomics.php

```diff
@@ -0,0 +1,55 @@
+<?php
+require_once __DIR__ . '/../../../init.php';
+require_once __DIR__ . '/../../../includes/gatewayfunctions.php';
+require_once __DIR__ . '/../../../includes/invoicefunctions.php';
+require_once __DIR__ . '/blockonomics.php';
+
+use Blockonomics\Blockonomics;
+
+
+// Init Blockonomics class
+$blockonomics = new Blockonomics();
+$systemUrl = \App::getSystemURL();
+
+function logMessage($message) {
+    $logFile = __DIR__ . '/poll_log.txt';
+    file_put_contents($logFile, date('Y-m-d H:i:s') . ' - ' . $message . PHP_EOL, FILE_APPEND);
+}
+
+function process_finish_order($finish_order, $crypto, $txid)
+{
+    global $blockonomics;
+    require_once $blockonomics->getLangFilePath();
+
+    if ($crypto === "usdt") {
+
+        $order = $blockonomics->processOrderHash($finish_order, $crypto);
+
+        if (empty($order)) {
+            return;
+        }
+
+        $any_existing_order = $blockonomics->getOrderByTransaction($txid);
+
+        if ($any_existing_order && $any_existing_order["order_id"]) {
+            return;
+        }
+
+        $invoiceId = $order->id_order;
+        $new_address = $crypto . '-' . $invoiceId;
+        
+        $subdomain = $blockonomics->getNetworktype() === "Test" ? "sepolia" : "www";
+
+        $blockonomics_currency = $blockonomics->getSupportedCurrencies()[$crypto];
+
+        $invoiceNote = '<b>' . $_BLOCKLANG['invoiceNote']['waiting'] . ' <img src="' . $systemUrl . 'modules/gateways/blockonomics/assets/img/usdt.png" style="max-width: 20px;"> ' . $blockonomics_currency->name . ' ' . $_BLOCKLANG['invoiceNote']['network'] . "</b>\r\r" .
+        $blockonomics_currency->name . " transaction id:\r" .
+            '<a target="_blank" href="https://' . $subdomain . ".etherscan.io/tx/$txid\">$txid</a>";
+
+        $blockonomics->updateOrderInDb($new_address, $txid, 0, 0);
+        $blockonomics->updateInvoiceNote($invoiceId, $invoiceNote);
+
+        $path = __DIR__ . '/pollTransactionStatus.php';
+        exec("php $path $txid > /dev/null &", $output, $return_var);
```

Comment:
Using `exec` and `&` will open a new background process. This will have implications on server resources, especially when lots of payments are still waiting for confirmation. This should be fine for the pilot implementation until USDT has been added to Blockonomics backend, although, we should ensure we do not leave any background processes running. eg. payments with low fee which take long to confirm or sites with high volume of payments.